### PR TITLE
E2E tests for kubectl label command

### PR DESCRIPTION
**Pull request by janetkuo**
#10764

cc @bgrant0607 @jlowdermilk @mwielgus @krousey @lavalamp @davidopp

### CODE REVIEW

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -202,6 +202,42 @@ var _ = Describe("Kubectl client", func() {
...
+			By("creating the pod")
+			nsFlag = fmt.Sprintf("--namespace=%v", ns)
+			runKubectl("create", "-f", podPath, nsFlag)
+			checkPodsRunningReady(c, ns, []string{simplePodName}, podStartTimeout)
+		})
+		AfterEach(func() {
+			cleanup(podPath, ns, simplePodSelector)
+		})
+
+		It("should check if the label can be updated on a resource", func() {
```
Comment (jlowdermilk) [inline test/e2e/kubectl.go]: Nit: "should update the label on a resource".

File: test/e2e/kubectl.go [hunk 2]
```diff
@@ -202,6 +202,42 @@ var _ = Describe("Kubectl client", func() {
...
+			labelName := "testing-label"
+			labelValue := "testing-label-value"
+
+			By("adding the label " + labelName + " with value " + labelValue + " to a pod")
+			runKubectl("label", "pods", simplePodName, labelName+"="+labelValue, nsFlag)
+			By("checkintg if the pod has the label " + labelName + " with the value " + labelValue)
```
Comment (jlowdermilk) [inline test/e2e/kubectl.go]: s/checkintg if/verifying/

File: test/e2e/kubectl.go [hunk 3]
```diff
@@ -202,6 +202,42 @@ var _ = Describe("Kubectl client", func() {
...
+			output := runKubectl("get", "pod", simplePodName, "-L", labelName, nsFlag)
+			if !strings.Contains(output, labelValue) {
+				Failf("Failed updating label " + labelName + " to the pod " + simplePodName)
+			}
+
+			By("removing the label " + labelName + " of a pod")
+			runKubectl("label", "pods", simplePodName, labelName+"-", nsFlag)
+			By("checkintg if the pod doesn't have the label " + labelName)
```
Comment (jlowdermilk) [inline test/e2e/kubectl.go]: same as above

File: test/e2e/kubectl.go (hunk 1 above)
Comment (JanetKuo) [inline test/e2e/kubectl.go]: Fixed

File: test/e2e/kubectl.go (hunk 2 above)
Comment (JanetKuo) [inline test/e2e/kubectl.go]: Fixed

File: test/e2e/kubectl.go (hunk 3 above)
Comment (JanetKuo) [inline test/e2e/kubectl.go]: Fixed

**Comment by JanetKuo** [comment]
PTAL. Thanks!

**Comment by jlowdermilk** [comment]
LGTM. Thanks @JanetKuo

**Comment by JanetKuo** [comment]
Thanks @jlowdermilk. Should we wait until next week to merge?

**Comment by jlowdermilk** [comment]
We've been marking the other kubectl e2e tests as post-1.0 so I think we're waiting? @bgrant0607, @brendandburns to confirm. Imo, would be good to get tests for #10764 in asap. That being said unless the tests uncover p0 bugs, probably fine to wait until next week to merge, so long as we push them to the front of the queue so we can prevent regressions from all the other pending kubectl changes that are marked post-1.0.

**Comment by davidopp** [comment]
LGTM

I can't see any reason not to merge this?

**Comment by jlowdermilk** [comment]
@davidopp thanks for the goahead. If you have time, there are a number of other open prs that add e2e coverage of kubectl: #10920, #11006, #10990. Not all are ready to merge, but if you are willing ok the merge it would be great to get them in.

There's also proxy e2e tests (#10813, #10608), but those involve code changes, so might be better to wait.

**Pull request merged**